app/controllers/api/graphql_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module API
+  class GraphqlController < ActionController::API
+    # If accessing from outside this domain, nullify the session
+    # This allows for outside API access while preventing CSRF attacks,
+    # but you'll have to authenticate your user separately
+    # protect_from_forgery with: :null_session
+
+    def execute
+      render json: result
+    rescue StandardError => e
+      raise e unless Rails.env.development?
+
+      handle_error_in_development e
+    end
+
+    private
+
+    def result
+      variables = ensure_hash(params[:variables])
+      query = params[:query]
+      operation_name = params[:operationName]
+      context = {
+        # Query context goes here, for example:
+        # current_user: current_user,
+      }
+      NimbleSurveyWebSchema.execute(query, variables: variables, context: context, operation_name: operation_name)
+    end
+
+    # Handle form data, JSON body, or a blank value
+    def ensure_hash(ambiguous_param)
+      case ambiguous_param
+      when String
+        ambiguous_param.present? ? ensure_hash(JSON.parse(ambiguous_param)) : {}
+      when Hash, ActionController::Parameters
+        ambiguous_param
+      when nil
+        {}
+      else
+        raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
+      end
+    end
+
+    def handle_error_in_development(exception)
+      logger.error exception.message
+      logger.error exception.backtrace.join("\n")
+
+      render json: {
+        errors: [{ message: exception.message, backtrace: exception.backtrace }],
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/controllers/api/graphql_controller.rb#L50"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>